static/src/javascripts/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -0,0 +1,263 @@
+    // behaves as usual.
+    const surveyToShow = local.get('surveyToShow');
+
+    if (surveyToShow) {
+        queryParams.surveyToShow = surveyToShow;
+    }
+
+    // get the list of surveys that can't be shown as they have been shown recently
+    const surveysNotToShow = getSurveyIdsNotToShow();
+
+    if (surveysNotToShow) {
+        queryParams.surveysNotToShow = surveysNotToShow;
+    }
+
+    return tailor.getSuggestedSurvey(queryParams).then(suggestion => {
+        if (suggestion) {
+            storeSurveyShowedInCookie(suggestion.data);
+
+            const json = getJsonFromSurvey(suggestion.data.survey);
+            const survey = bonzo.create(template(tailorSurveyHtml, json));
+
+            return inArticleWriter(survey, suggestion.data.survey.surveyId);
+        }
+        Promise.resolve();
+    });
+};
+
+const disableRadioButtons = buttonClassName => {
+    const radioButtons = document.getElementsByClassName(buttonClassName);
+    bonzo(radioButtons).each(button => {
+        // eslint-disable-next-line no-param-reassign
+        button.disabled = true;
+    });
+};
+
+const surveyFadeOut = () => {
+    const surveyContent = document.getElementsByClassName(
+        'impressions-survey__content'
+    );
+
+    surveyContent[0].classList.add('js-impressions-survey__fadeout');
+};
+
+const thankyouFadeIn = () => {
+    const surveyThanks = document.getElementsByClassName(
+        'impressions-survey__thanks'
+    );
+
+    surveyThanks[0].classList.add('js-impressions-survey__fadein');
+};
+
+const recordOphanAbEvent = (answer, surveyId) => {
+    const componentId = `data_tailor_survey_${surveyId}`;
+
+    ophan.record({
+        component: componentId,
+        value: answer,
+    });
+};
+
+const handleSurveyResponse = surveyId => {
+    const surveyQuestions = [
+        ...document.getElementsByClassName('fi-survey__button'),
+    ];
+
+    surveyQuestions.forEach(question => {
+        bean.on(question, 'click', event => {
+            if (event.target.attributes.getNamedItem('data-link-name')) {
+                const answer = event.target.attributes.getNamedItem(
+                    'data-link-name'
+                ).value;
+
+                recordOphanAbEvent(answer, surveyId);
+
+                mediator.emit('tailor:survey:clicked');
+                fastdom.write(() => {
+                    disableRadioButtons('fi-survey__button');
+                    surveyFadeOut();
+                    thankyouFadeIn();
+                });
+            }
+        });
+    });
+};
+
+class TailorSurvey {
+    id: string;
+    start: string;
+    expiry: string;
+    author: string;
+    description: string;
+    audience: number;
+    audienceOffset: number;
+    successMeasure: string;
+    audienceCriteria: string;
+    dataLinkNames: string;
+    idealOutcome: string;
+    variants: Array<Object>;
+    canRun: () => boolean;
+
+    constructor() {
+        this.id = 'TailorSurvey';
```

Comment:
What do you think about writing this as a pure Object

`const props = { id: '...' }` and assign it later as a whole to this `Object.assign(this, props)`? Less of the annoying and repetitive `this.` ..